internal/host/transition.go

```diff
@@ -439,6 +439,8 @@ func (th *transitionHandler) HasInstallationInProgressTimedOut(sw stateswitch.St
 // Return a post transition function with a constant reason
 func (th *transitionHandler) PostRefreshHost(reason string) stateswitch.PostTransition {
 	ret := func(sw stateswitch.StateSwitch, args stateswitch.TransitionArgs) error {
+		// Not using reason directly to avoid closures issue.
+		r := reason
```

Comment:
maybe calling it template will be a more proper name (sorry - didn't think about it in the first review)